README.md

```diff
@@ -1,3 +1,337 @@
+
+### Windows binaries
+
+Windows needs pre-built binary DLLs. The expected workflow is that the release is made, then some kind of build takes
+place (for example, in a GitHub Action, or manually on a compatible build environment), and the `.dll` file is added to
+the GitHub release.
+
+The name for the DLLs must follow the following pattern:
+
+ * `php_{extension-name}-{tag}-{php-maj/min}-{compiler}{"-nts"}?{-arch}?.dll`
+
+The descriptions of these items:
+
+ * `extension-name` the name of the extension, e.g. `xdebug` - this is the same as defined in `composer.json` minus `ext-`
+ * `tag` for example `3.3.0alpha3` - defined by the tag/release you have made
+ * `php-maj/min` - for example `8.3` for PHP 8.3.*
+ * `compiler` - usually `vc16`?
+ * `-nts` - optional - non-thread safe. If omitted, thread-safe mode is assumed
+ * `-arch` - optional - for example `x86_64`, fetch using `uname -i`. If omitted, the extension is assumed to be
+   compatible with all architectures
+
+### Notes on the `composer.json`
+
+ * The `name` is not a typical `vendor/package` format, it is `ext-package` format.
+ * The `type` dictates if it is a PHP Module (`php-ext`) or a Zend Extension (`php-ext-zend`).
+ * Typically, there will be almost no `require` definitions, except `php` version itself
+ * The `php-ext` is a new top-level element to provide additional metadata for building the extension, if required.
+   * `priority` can be used to prefix to the INI file, e.g. `90-xdebug.ini` which affects the loading order
+   * `config` may contain flags passed to the compilation of the ext (?)
+
+## End user: installing an npecl package
+
+Run:
+
+```bash
+$ npecl install ext-xdebug
+```
+
+npecl uses `composer/composer` library to help resolve dependencies
+
+```mermaid
+sequenceDiagram
+    participant npecl
+    participant composer as composer library
+    participant packagist as packagist.org
+
+    npecl ->> composer : request to install "ext-xdebug"
+    composer ->> packagist : read metadata to resolve dependencies
+    packagist ->> composer : metadata
+    composer ->> npecl : release information
+```
+
+ * if a downstream dep (e.g. `ext-zlib` in the `ext-xdebug` example above) is not installed, Composer can detect this
+   missing depenedency, and warn accordingly.
+
+Once we have the release information, for Linux:
+
+### non-Windows installation
+
+ * download/extract the source tarball
+ * run `phpize`
+ * run `./configure <options>`
+ * run `make`
+ * run `make install` (note: this part may need `sudo`)
+
+### Windows installation
+
+Determine the expected name for the Windows DLL:
+
+ * `extension-name` - We know this already minus `ext-` from the package name
+ * `tag` - Composer gave us the release version
+ * `php-maj/min` - We know this from the version of PHP that invoked `npecl`
+ * `compiler` - ?
+ * `-nts` or omitted - We know this from the version of PHP that invoked `npecl`
+ * `arch` - find using `uname -i`
```

Comment:
`uname -i` isn't something you can run on Windows. There is this:

```
D:\PHP>uname -m
i686
```

But that doesn't say it's 64-bit :-)